### PR TITLE
Let 800 fe invited user

### PR DESCRIPTION
**Pull request by barbara-chaves**
This PR fixes 3 bugs of invite users flow

The 3 tasks are in the same PR because they are related and the whole flow can be tested at once

## Testing instructions

The BUGS were:

### 1. **Show error if invitation token to a PD/investor account is invalid** 
  If the user opens a link with an invalid token, the user is never shown an error. Instead, the loading icon keeps spinning.
  Bug reproduction steps:
  - Open https://staging.hecoinvest.org/invitation?invitation_token=jhojoijoij
  - You will see an error is never shown.
  
**FIX**: A message of 'Invalid invitation token' is shown 

### 2. **Show invited user in the PD/investor's users table right away**
  After a user is invited to an account, the users table is not refreshed to show that new user.
  Bug reproduction steps:

- Log in to an account with the owner user
- Go to the “Users” tab in the dashboard
- Invite a new user by clicking on “Invite user”
- Send the invitation
- After the modal has closed, the table do not show that new user.

**FIX**: The table is refreshed and the invited users are displayed

### 3. **Invited user to a PD/investor account shown an error instead of being redirected to the Dashboard**
  When a user is invited to an account, after they’ve completed the sign up page, they are not redirected to the Dashboard but shown an “Invalid invitation token” error instead. 

  Bug reproduction steps:
  - Sign in with the owner user of an account
  - Invite a non-existing user with an email address that you own
  - Click the invitation link from the email
  - Complete the sign up page
  - After submitting the form, you are presented with an error instead of being redirected to the Dashboard.
 
 **FIX**: The user is redirected to the dashboard

## Tracking

[Show error if invitation token to a PD/investor account is invalid](https://vizzuality.atlassian.net/browse/LET-798)
[Show invited user in the PD/investor's users table right away](https://vizzuality.atlassian.net/browse/LET-799)
[Invited user to a PD/investor account shown an error instead of being redirected to the Dashboard](https://vizzuality.atlassian.net/browse/LET-800)

### CODE REVIEW

File: frontend/containers/dashboard/users/table/component.tsx [hunk 1]
```diff
@@ -23,13 +23,13 @@ export const UsersTable: FC<UsersTableProps> = ({ isOwner }) => {
   const queryParams = useQueryParams();
 
   const {
-    data: { data: users } = { data: [] },
+    data: { data: users },
```
Comment (SARodrigues) [inline frontend/containers/dashboard/users/table/component.tsx:26]:
```suggestion
    data: { data: users } = { data: [] },
```

Comment (SARodrigues) [reply frontend/containers/dashboard/users/table/component.tsx:26]: This removal breaks the page for me, especially when doing a refresh.
users

**Review by SARodrigues — CHANGES_REQUESTED** [review]
Following the testing instructions everything works great with the exception of visiting the site on a fresh tab (or doing a hard refresh), where the users' table breaks.

File: frontend/containers/dashboard/users/table/component.tsx (hunk 1 above)
Comment (barbara-chaves) [reply frontend/containers/dashboard/users/table/component.tsx]: I changed it for the exported {users} const

**Review by SARodrigues — APPROVED** [review]

**Pull request merged**